File: Python_Virtual_World/Fox.py
from MyVector import MyVector
from Animal import Animal


class Fox(Animal):


    STRENGTH = 3
    INITIATIVE = 7


    def __init__(self, position: MyVector):
        super().__init__(position, Fox.STRENGTH, Fox.INITIATIVE)


    def isSmelling(self):

        return True


    def draw(self) -> str:

        return "orange"


    def __str__(self):
        return "Fox"

File: Java_Virtual_World/src/Gameplay/Animals/Fox.java
package Gameplay.Animals;

import Gameplay.Organism;

public class Fox extends Animal {
    public static final int STRENGTH = 3;
    public static final int INITIATIVE = 7;

    public Fox(int x, int y){
        super(x ,y , STRENGTH, INITIATIVE);
    }

    @Override
    public boolean isSmelling(){
        return true;
    }
    @Override
    public String Draw(){
        return "🦊";
    }
    @Override
    public Organism Copy(){
        return new Fox(x,y);
    }
    @Override
    public String toString(){
        return "Fox";
    }
}
